api_app/analyzers_manager/file_analyzers/cape_sandbox.py

```diff
@@ -0,0 +1,232 @@
+                the following code further fetches it's information.
+                response_json in this case should look somewhat like this:
+
+            {
+                'error': True,
+                'error_value': 'Error adding task to database',
+                'errors': [{
+                    'filename.exe':
+                        'Not unique, as unique option set on submit or in conf/web.conf'
+                }]
+            }
+            """
+
+            logger.info(
+                f"[INFO] (Job: {self.job_id} -> "
+                "File uploaded is already present in the database. "
+                "Querying its information through it's md5 hash.."
+            )
+
+            status_id = self.__search_by_md5()
+            gui_report_url = self.__base_url + "/submit/status/" + status_id
+            report_url = (
+                self.__base_url + "/apiv2/tasks/get/report/" + status_id + "/json"
+            )
+            to_respond["result_url"] = gui_report_url
+
+            try:
+                final_request = self.__session.get(
+                    report_url,
+                )
+            except requests.RequestException as e:
+                raise AnalyzerRunException(e)
+
+            final_json = final_request.json()
+            to_respond["response"] = final_json
+
+            response_json = to_respond
+
+        return response_json
+
+    def __search_by_md5(self) -> str:
+        db_search_url = self.__base_url + "/apiv2/tasks/search/md5/" + self.md5
+
+        try:
+            q = self.__session.get(db_search_url)
+            q.raise_for_status()
+
+        except requests.RequestException as e:
+            raise AnalyzerRunException(e)
+
+        data_list = q.json().get("data")
+        if data_list:
+            status_id_int = data_list[0].get("id")
+            status_id = str(status_id_int)
+            return status_id
+
+        raise AnalyzerRunException(
+            "'data' key in response isn't populated as expected."
+        )
+
+    def __poll_for_result(
+        self,
+        task_id,
+    ) -> dict:
+        status_api = self.__base_url + "/apiv2/tasks/status/" + str(task_id)
+        for i in range(self.max_tries):
+            logger.info(
+                f"[POLLING] (Job: {self.job_id} -> "
+                f"CAPEsandbox __poll_for_result #{i + 1}/{self.max_tries}"
+            )
+            try:
+                r = self.__session.get(status_api)
+                if r.status_code == 429:
+                    logger.info(
+                        f"[INFO] (Job: {self.job_id} -> "
+                        "Rate limited by CAPESandbox API. "
+                        f"Sleeping for {self.poll_distance} seconds."
+                    )
+                    time.sleep(self.poll_distance)
+                else:
+                    r.raise_for_status()
+            except requests.RequestException as e:
+                if i == self.max_tries - 1:
```

Comment:
honestly I do not like such kind of inline calculation. This could better be handled by a bool flag that is checked at the end of the loop